### ISSUE TITLE
TS Server fatal error:  Debug Failure. False expression.

### ISSUE THREAD

**Issue by lachopysparks** [microsoft/TypeScript#49986]
Issue Type: <b>Bug</b>

❗️❗️❗️ Please fill in the sections below to help us diagnose the issue ❗️❗️❗️

**TypeScript Version:** 4.7.3

**Steps to reproduce crash**
1.
2.
3.

**TS Server Log**
❗️Server logging disabled. To help us fix crashes like this, please enable logging by setting:
```json
"typescript.tsserver.log": "verbose"
```
After enabling this setting, future crash reports will include the server log.

**TS Server Error Stack**
Server: `semantic`
```
Error: Debug Failure. False expression.
    at _loop_3 (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:171897:34)
    at ProjectService.assignOrphanScriptInfoToInferredProject (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:171905:25)
    at ProjectService.assignProjectToOpenedScriptInfo (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:173574:26)
    at c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:173848:97
    at Object.flatMap (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:508:25)
    at ProjectService.applyChangesInOpenFiles (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:173848:41)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:174983:46)
    at c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:177172:88
    at IpcIOSession.Session.executeWithRequestId (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:177163:28)
    at IpcIOSession.Session.executeCommand (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:177172:33)
    at IpcIOSession.Session.onMessage (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:177198:35)
    at process.<anonymous> (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:179833:31)
    at process.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
```

VS Code version: Code 1.69.2 (3b889b090b5ad5793f524b5d1d39fda662b96a2a, 2022-07-18T16:12:52.460Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 3600 6-Core Processor               (12 x 3950)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.89GB (11.94GB free)|
|Process Argv|--crash-reporter-id 91d1d981-6f5c-4015-9d02-9664e865eaf5|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (44)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.0
ng-template|Ang|14.0.1
vscode-django|bat|1.10.0
vscode-node-readme|ben|3.0.2
npm-intellisense|chr|1.4.2
path-intellisense|chr|2.8.1
dart-code|Dar|3.44.0
flutter|Dar|3.44.0
vscode-markdownlint|Dav|0.47.0
vscode-eslint|dba|2.2.6
gitlens|eam|12.1.2
prettier-vscode|esb|9.5.0
copilot|Git|1.33.6266
vscode-pull-request-github|Git|0.46.0
go|gol|0.35.1
gc-excelviewer|Gra|4.2.55
todo-tree|Gru|0.0.215
vscode-drawio|hed|1.6.4
generate-index|Jay|1.6.0
rainbow-csv|mec|2.4.0
dotenv|mik|1.0.1
vscode-docker|ms-|1.22.1
vscode-ros|ms-|0.8.3
python|ms-|2022.11.12001005
vscode-pylance|ms-|2022.7.30
jupyter|ms-|2022.6.1201981810
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.8
remote-containers|ms-|0.241.3
remote-ssh|ms-|0.84.0
remote-ssh-edit|ms-|0.80.0
remote-wsl|ms-|0.66.3
cpptools|ms-|1.10.8
makefile-tools|ms-|0.5.0
powershell|ms-|2022.6.3
angular-console|nrw|17.18.8
indent-rainbow|ode|8.3.1
material-icon-theme|PKi|4.19.0
vscode-xml|red|0.21.2022062916
LiveServer|rit|5.7.5
code-spell-checker|str|2.3.1
vscode-counter|uct|3.0.5
intellicode-api-usage-examples|Vis|0.1.2
vscodeintellicode|Vis|1.2.22

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonvspyt551:30345470
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
vscaat:30438848
pylanb8912cf:30529770
vsclangdc:30486549

```

</details>

**Comment by DanielRosenwasser** [comment]
We believe that the last comment's errors should be fixed in TypeScript nightly thanks to https://github.com/microsoft/TypeScript/pull/50042.

[If you use the nightly TypeScript extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next), do you still see the same behavior? If so, we can set up a private way to send logs over.

**Comment by distante** [comment]
> We believe that the last comment's errors should be fixed in TypeScript nightly thanks to #50042.
> 
> [If you use the nightly TypeScript extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next), do you still see the same behavior? If so, we can set up a private way to send logs over.

It is working for me using the nightly version.

**Comment by RyanCavanaugh** [comment]
Thanks!
